docs/tutorials/build-your-first-blazor-app.md

```diff
@@ -0,0 +1,485 @@
+
+    ![Blazor count by ten](https://msdnshared.blob.core.windows.net/media/2018/03/blazor-home-count-by-ten.png)
+
+## Route to components
+
+The `@page` directive at the top of the *Counter.cshtml* file specifies that this component is a page to which requests can be routed. Specifically, the `Counter` component handles requests sent to `/Counter`. Without the `@page` directive, the component wouldn't handle routed requests, but the component could still be used by other components.
+
+## Dependency injection
+
+Services registered with the app's service provider are available to components via [dependency injection (DI)](https://docs.microsoft.com/aspnet/core/fundamentals/dependency-injection). Services can be injected into a component using the `@inject` directive.
+
+Take a look at the implementation of the `FetchData` component in *FetchData.cshtml*. The `@inject` directive is used to inject an [HttpClient](https://docs.microsoft.com/dotnet/api/system.net.http.httpclient) instance into the component.
+
+```cshtml
+@page "/fetchdata"
+@inject HttpClient Http
+```
+
+The `FetchData` component uses the injected `HttpClient` to retrieve JSON data from the server when the component is initialized. Under the covers, the `HttpClient` provided by the Blazor runtime is implemented using JavaScript interop to call the underlying browser's Fetch API to send the request (from C#, it's possible to call any JavaScript library or browser API). The retrieved data is deserialized into the `forecasts` C# variable as an array of `WeatherForecast` objects.
+
+```cshtml
+@functions {
+    WeatherForecast[] forecasts;
+
+    protected override async Task OnInitAsync()
+    {
+        forecasts = await Http.GetJsonAsync<WeatherForecast[]>("/sample-data/weather.json");
+    }
+
+    class WeatherForecast
+    {
+        public DateTime Date { get; set; }
+        public int TemperatureC { get; set; }
+        public int TemperatureF { get; set; }
+        public string Summary { get; set; }
+    }
+}
+```
+
+A `@foreach` loop is used to render each forecast instance as a row in the weather table.
+
+```cshtml
+<table class="table">
+    <thead>
+        <tr>
+            <th>Date</th>
+            <th>Temp. (C)</th>
+            <th>Temp. (F)</th>
+            <th>Summary</th>
+        </tr>
+    </thead>
+    <tbody>
+        @foreach (var forecast in forecasts)
+        {
+            <tr>
+                <td>@forecast.Date.ToShortDateString()</td>
+                <td>@forecast.TemperatureC</td>
+                <td>@forecast.TemperatureF</td>
+                <td>@forecast.Summary</td>
+            </tr>
+        }
+    </tbody>
+</table>
+```
+
+## Build a todo list
+
+Add a new page to the app that implements a simple todo list.
+
+1. Add a Todo page to the app by right-clicking on the *Pages* folder and selecting **Add** > **New item** > **Razor View**. Name the file *Todo.cshtml* and select **OK**.
```

Comment:
Let's not use the item templates, but instead just tell people to add a *Todo.cshtml* file to the *Pages* folder. In usability testing folks pretty easily get lost in the item template dialog.